tools/check_strings.py

```diff
@@ -0,0 +1,2549 @@
+    # ]
+    # # Inner funcs
+    # for node in function_definitions:
+    #     for item in node.body:
+    #         if isinstance(item, ast.FunctionDef):
+    #             function_definitions.append(item)
+
+    docstrings.extend([ast.get_docstring(f) for f in function_definitions])
+    class_definitions = [
+        node for node in module.body if isinstance(node, ast.ClassDef)
+    ]
+    docstrings.extend([ast.get_docstring(f) for f in class_definitions])
+    method_definitions = []
+
+    for class_def in class_definitions:
+        method_definitions.extend(
+            [
+                node
+                for node in class_def.body
+                if isinstance(node, ast.FunctionDef)
+            ]
+        )
+
+    docstrings.extend([ast.get_docstring(f) for f in method_definitions])
+    docstrings.append(ast.get_docstring(module))
+    docstrings = [doc for doc in docstrings if doc]
+
+    results = {}
+    for doc in docstrings:
+        key = " ".join([it for it in doc.split() if it != ""])
+        results[key] = doc
+
+    return results
+
+
+def find_strings(fpath: str):
+    """
+    Find all strings (and f-strings) for the given file.
+
+    Parameters
+    ----------
+    fpath : str
+        File path.
+
+    Returns
+    -------
+    tuple
+        A tuple of tuples for the line number and the string found.
+    """
+    strings = {}
+    with open(fpath) as f:
+        for toktype, tokstr, (lineno, _), _, _ in tokenize.generate_tokens(
+            f.readline
+        ):
+            if toktype == tokenize.STRING:
+                try:
+                    string = eval(tokstr)
+                except Exception:
+                    string = eval(tokstr[1:])
+
+                if isinstance(string, str):
+                    key = " ".join([it for it in string.split() if it != ""])
+                    strings[(lineno, key)] = (lineno, string)
+
+    return strings
+
+
+def find_trans_strings(fpath: str):
+    """
+    Find all translation strings for the given file.
+
+    Parameters
+    ----------
+    fpath : str
+        File path.
+
+    Returns
+    -------
+    """
+    with open(fpath) as fh:
+        data = fh.read()
+
+    module = ast.parse(data)
+    trans_strings = {}
+    show_trans_strings.visit(module)
+    for string in show_trans_strings._found:
+        key = " ".join([it for it in string.split()])
+        trans_strings[key] = string
+
+    show_trans_strings._found = set()
+
+    return trans_strings
+
+
+def import_module_by_path(fpath):
+    import importlib.util
+
+    fpath = fpath.replace("\\", "/")
+    module_name = fpath.replace(".py", "").replace("/", ".")
+    try:
+        spec = importlib.util.spec_from_file_location(module_name, fpath)
+        module = importlib.util.module_from_spec(spec)
+        spec.loader.exec_module(module)
+    except Exception:
+        module = None
+
+    return module
+
+
+def find_untranslated_strings(paths, skip_words):
```

Comment:
Doc string, also including what get's raised. Am I right in thinking that if a doc string is on the skip list but not found it will raise an issue? I think that will help us keep the list of skipped strings up to date too

Reply:
@sofroniewn nothing gets raised here, we collect them and only do a `sys.exit(1)` which should fail on CI so that we can provide a full list of issues.

Added docs.